opennlp-tools/src/main/java/opennlp/tools/lemmatizer/DictionaryLemmatizer.java

```diff
@@ -47,26 +47,34 @@
    * @param dictionary
    *          the input dictionary via inputstream
    */
+  // To have duplicate keys we need to have a collection for values too,
+  // this way:
+  // 1. We could get every lemma for a word,pos pair in the key
+  // 2. We could get every pos,lemma for a word in the key
+  // Crucially, both keys and values need to be collections, probably lists
   public DictionaryLemmatizer(final InputStream dictionary) {
```

Comment:
The constructor should throw an IOException. The caller has to deal with the IOException that can be thrown by reading from the stream.